Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript

```diff
@@ -2992,11 +2992,33 @@ command scriptFormat pScope, pDontGroup
       local tOldText
       put the text of field "Script" of me into tOldText
       
+      local tFirstLine, tLastLine, tCharsAboveFirstLineOld, tCharsAboveLastLineOld
+      get the selectedLine
+      put word 2 of it into tFirstLine
+      if word 4 of it is a number then put word 4 of it into tLastLine else put tFirstLine into tLastLine
+      
+      put the number of chars of line 1 to tFirstLine of tOldText into tCharsAboveFirstLineOld
+      put the number of chars of line 1 to tLastLine of tOldText into tCharsAboveLastLineOld
```

Comment:
@BerndN this is likely to be slow for long scripts. Could you add something like:
```
if tFirstLine is tLastLine then
    put tCharsAboveFirstLineOld into tCharsAboveLastLineOld
else
    put the number of chars of line 1 to tLastLine of tOldText into tCharsAboveLastLineOld
end if
```
Same below

Also it might be worth experimenting with `charIndex` for speed:

`put the charIndex of line tFirstLine of tOldText`